core/src/main/kotlin/fr/sncf/osrd/stdcm/graph/PostProcessingSimulation.kt

```diff
@@ -0,0 +1,314 @@
+                // Mareco allowances must have a non-zero capacity speed limit,
+                // which may cause "too much time" errors.
+                // We can ignore this exception and move on to the linear allowance as fallback
+                logger.info("Can't slow down enough to match the given standard allowance")
+                break
+            } else throw e
+        }
+    }
+    if (!isMareco) {
+        throw RuntimeException(
+            "Failed to compute a standard allowance that wouldn't cause conflicts"
+        )
+    } else {
+        logger.info("Failed to compute a mareco standard allowance, fallback to linear allowance")
+        return buildFinalEnvelope(
+            graph,
+            maxSpeedEnvelope,
+            ranges,
+            standardAllowance,
+            envelopeSimPath,
+            rollingStock,
+            timeStep,
+            comfort,
+            blockAvailability,
+            departureTime,
+            stops,
+            false,
+        )
+    }
+}
+
+/** Initialize all fixed points at stop locations, including stop durations. */
+private fun initFixedPoints(
+    ranges: List<EdgeRange<STDCMEdge, STDCMEdge>>,
+    stops: List<TrainStop>,
+    departureTime: Double,
+    length: Length<TravelledPath>,
+    hasStandardAllowance: Boolean,
+): TreeSet<FixedTimePoint> {
+    val res = TreeSet<FixedTimePoint>()
+    var prevStopTime = 0.0
+    for (stop in stops) {
+        res.add(
+            makeFixedPoint(
+                ranges,
+                Offset(Distance.fromMeters(stop.position)),
+                departureTime,
+                stop.duration
+            )
+        )
+        prevStopTime += stop.duration
+    }
+    if (hasStandardAllowance && res.none { it.offset == length })
+        res.add(makeFixedPoint(ranges, length, departureTime, 0.0))
+    return res
+}
+
+/**
+ * Create a new fixed point at a given offset. The reference time is fetched on the given ranges.
+ */
+private fun makeFixedPoint(
+    ranges: List<EdgeRange<STDCMEdge, STDCMEdge>>,
+    conflictOffset: Offset<TravelledPath>,
+    departureTime: Double,
+    stopDuration: Double = 0.0,
+): FixedTimePoint {
+    return FixedTimePoint(
+        getTimeOnRanges(ranges, conflictOffset, departureTime),
+        conflictOffset,
+        if (stopDuration > 0) stopDuration else null
+    )
+}
+
+/**
+ * Returns the time expected during the exploration at the given offset. The returned value is an
+ * offset compared to the train departure time.
+ */
+private fun getTimeOnRanges(
+    ranges: List<EdgeRange<STDCMEdge, STDCMEdge>>,
+    offset: Offset<TravelledPath>,
+    departureTime: Double,
+): Double {
+    var remainingDistance = offset.distance
+    for (range in ranges) {
+        assert(range.start.distance == 0.meters)
+        val edge = range.edge
+        if (remainingDistance <= range.end.distance) {
+            val absoluteTime = edge.getApproximateTimeAtLocation(Offset(remainingDistance))
```

Comment:
After discussion with @eckter : might be better to create a `FixedTimePoint` on an edge transition, where we know the exact time, rather than find the approximate time at which the train would be at the edge's offset with a linear interpolation (as done in `getAppoximateTimeAtLocation`).